### ISSUE TITLE
Tree View lacks accessible names

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3570]
## Describe the bug

The treenode buttons lack accessible names like the one below lack accessible names. Screen reader users will not understand the purpose of these buttons when they read them.

The Tree Views lack accessible names. Screen reader users will not know the purpose of the tree controls. 

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Using a screen reader, read the Tree View - Basic control and observe there is no accessible name announced for it. 
- Read one of the treenode buttons and notice that the screen reader does not announce an accessible name that describes the button.

## Expected behavior

- Either provide a mechanism or an example for the consumer to add an accessible name to a tree using aria-label or aria-labelledby
- Follow the implementation guidance at https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

**Comment by gnomeontherun** [comment]
I'm going to close this as a duplicate since it will be resolved by #3569
